src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -5017,6 +5017,59 @@ namespace ts.projectSystem {
             service.openExternalProject({ projectFileName: "p", rootFiles: [toExternalFile(f1.path)], options: { importHelpers: true } });
             service.checkNumberOfProjects({ externalProjects: 1 });
         });
+
+        it("should not crash when getting code fix from dynamic import", () => {

```

Comment:
This issue didn't have anything to do with the server, just a bug in a codefix, so would be better as a fourslash test. See the many existing `codeFixInferFromUsage...` fourslash tests.

Reply:
fourslash is not easily equipped right now to test the negative scenarios of the codeFix. The current error happens because one of the action's change is undefined and we try to map the change to code edit. I am not sure how i test that from fourslash with current infrastructure.

Reply:
I got a failing fourslash test in the `master` branch:
```ts
/// <reference path='fourslash.ts' />

// @noImplicitAny: true

// @Filename: /a.ts
////export class D {}
////export default new D();

// @Filename: /b.ts
////export class C {
////    [|set x(val) {}|]
////    method() { this.x = import("./a"); }
////}

goTo.file("/b.ts");
verify.not.codeFixAvailable();
```